### PR TITLE
ADD: added a file to interface with the NI USB DAQ

### PR DESCRIPTION
**Pull request by beegica**

### CODE REVIEW

File: smile/niusb_interface.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+from state import State
+from event import event_time
+from clock import clock
+from ref import NotAvailable
+
+try:
+    from nidaqmx.task import Task
+    _got_nidaqmx = True
+except ImportError:
+    print("Unable to import pylsl!")
```
Comment (psederberg) [inline smile/niusb_interface.py]: This should be nidaqmx, right?

File: smile/niusb_interface.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    _got_nidaqmx = False
+
+
+_ni_tasks = {}
+
+
+def init_task(task_name="TASKING", min_val=0.0, max_val=1.0,
```
Comment (psederberg) [inline smile/niusb_interface.py]: Can we call the default task_name something else?

File: smile/niusb_interface.py (hunk 1 above)
Comment (beegica) [reply smile/niusb_interface.py]: yes. let me fix that.

File: smile/niusb_interface.py (hunk 2 above)
Comment (beegica) [reply smile/niusb_interface.py]: ttl_pulse?

File: smile/niusb_interface.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        else:
+            self._pulse_on = None
+            self._pulse_off = None
+            self._ended = True
+            clock.schedule(self.leave)
+            clock.schedule(self.finalize)
+            return
+
+        # set the pulse time
+        time_err = (end - start)/2.
```
Comment (psederberg) [inline smile/niusb_interface.py]: Are you sure this is the correct time if it blocks?

File: smile/niusb_interface.py (hunk 2 above)
Comment (psederberg) [reply smile/niusb_interface.py]: Sure

File: smile/niusb_interface.py (hunk 3 above)
Comment (beegica) [reply smile/niusb_interface.py]: You are correct, I thought you wanted it this way from a previous conversation but I must be mistaken. Its a quick fix.

File: smile/niusb_interface.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+    def _callback(self):
+        # we've started
+        self._started = True
+
+        # push it outlet
+        global _got_nidaqmx
+        if _got_nidaqmx:
+            start = clock.now()
+            if type(self._push_vals == list):
```
Comment (psederberg) [inline smile/niusb_interface.py]: You should set a var called `to_push` based on this conditional *before* you set `start`. That way the timing will be more accurate.

File: smile/niusb_interface.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+if __name__ == "__main__":
+
+    from experiment import Experiment
+    from state import Wait, Parallel, Loop, Debug
+    from video import Label
+    exp = Experiment()
+
+    # Initialize the outlet
+    task1 = init_task(task_name="TASKING1", min_val=0.0, max_val=1.0,
```
Comment (psederberg) [inline smile/niusb_interface.py]: Can you call them TASK1 and TASK2. The TASKING seems odd to me.

**Pull request merged**